### PR TITLE
Sdc changes

### PR DESCRIPTION
**Pull request by daco101**
This PR
- uses data criteria models for source data criteria, and parses these directly from the hqmf xml
- removes data criteria which was a duplicated of source data criteria
- pulls most basic fields out of the hqmf xml rather than using the existing parser (which could eventually be removed)

Pull requests into cqm-parsers require the following. Submitter and reviewer should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable ("N/A") and :white_check_mark:.

- [x] Update Gemfile after cqm-models

**Submitter:**
- [x] This pull request describes why these changes were made.
- [x] Internal ticket for this PR: https://jira.mitre.org/browse/BONNIE-1836
- [x] Internal ticket links to this PR
- [x] Code diff has been done and been reviewed
- [x] Tests are included and test edge cases
- [x] Tests have been run locally and pass

**Reviewer 1:**

Name: @losborne 
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

**Reviewer 2:**

Name: @jbradl11 
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

### CODE REVIEW

File: lib/measure-loader/source_data_criteria_loader.rb [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    def modelize_data_criteria_entry(entry)
+      criteria = entry.at_xpath('./*[xmlns:templateId/xmlns:item]')
+      model_fields =  if entry.at_css('localVariableName').present?
+                        model_fields_for_standard_data_criteria(criteria, entry)
+                      else
+                        model_fields_for_single_code_reference_data_criteria(criteria)
+                      end
+      hqmf_template_oid = criteria.at_css('templateId/item')['root']
+      model = QDM::ModelFinder.by_hqmf_oid(hqmf_template_oid).new(model_fields)
+      model.description = model.hqmfTitle + ": " + model.description
```
Comment (losborne) [inline lib/measure-loader/source_data_criteria_loader.rb]:
```suggestion
      model.description = model.hqmfTitle + ': ' + model.description
```

File: lib/measure-loader/hqmf_measure_loader.rb [hunk 2]
```diff
@@ -4,26 +4,38 @@ module Measures
...
         measure.measure_period = hqmf_model_hash[:measure_period]
         measure.population_criteria = hqmf_model_hash[:population_criteria]
-        measure.source_data_criteria = hqmf_model_hash[:source_data_criteria]
-        measure.data_criteria = hqmf_model_hash[:data_criteria]
 
-        measure.measure_scoring = extract_measure_scoring(hqmf_xml)
-        measure.population_sets = extract_population_set_models(hqmf_xml, measure.measure_scoring)
-
-        # add observation info
+        # add observation info, note population_sets needs to have been added to the measure by now
```
Comment (losborne) [inline lib/measure-loader/hqmf_measure_loader.rb]: 🥇 good comment

File: lib/measure-loader/source_data_criteria_loader.rb [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+      source_data_criteria.uniq!(&:hqmfOid)
+      return source_data_criteria
+    end
+
+    private
+
+    def modelize_data_criteria_entry(entry)
+      criteria = entry.at_xpath('./*[xmlns:templateId/xmlns:item]')
+      model_fields =  if entry.at_css('localVariableName').present?
+                        model_fields_for_standard_data_criteria(criteria, entry)
```
Comment (losborne) [inline lib/measure-loader/source_data_criteria_loader.rb]:
```suggestion
                        extract_fields_from_standard_data_criteria(criteria, entry)
```

File: lib/measure-loader/source_data_criteria_loader.rb [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+                      else
+                        model_fields_for_single_code_reference_data_criteria(criteria)
+                      end
+      hqmf_template_oid = criteria.at_css('templateId/item')['root']
+      model = QDM::ModelFinder.by_hqmf_oid(hqmf_template_oid).new(model_fields)
+      model.description = model.hqmfTitle + ": " + model.description
+      return model
+    end
+
+    def model_fields_for_standard_data_criteria(criteria, entry)
```
Comment (losborne) [inline lib/measure-loader/source_data_criteria_loader.rb]:
```suggestion
    def extract_fields_from_standard_data_criteria(criteria, entry)
```

File: lib/measure-loader/source_data_criteria_loader.rb [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+    end
+
+    private
+
+    def modelize_data_criteria_entry(entry)
+      criteria = entry.at_xpath('./*[xmlns:templateId/xmlns:item]')
+      model_fields =  if entry.at_css('localVariableName').present?
+                        model_fields_for_standard_data_criteria(criteria, entry)
+                      else
+                        model_fields_for_single_code_reference_data_criteria(criteria)
```
Comment (losborne) [inline lib/measure-loader/source_data_criteria_loader.rb]:
```suggestion
                        extract_fields_from_single_code_reference_data_criteria(criteria)
```

File: lib/measure-loader/source_data_criteria_loader.rb [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+      entry_variable_name_prefix = entry.at_css('localVariableName')['value'].split('_').first
+      node_with_valueset = (criteria.at_css('value[valueSet]') || criteria.at_css('code[valueSet]'))
+      code_list_id = node_with_valueset.present? ? node_with_valueset['valueSet'] : nil
+      return {
+        description: entry_variable_name_prefix,
+        codeListId: code_list_id
+      }
+    end
+
+    def model_fields_for_single_code_reference_data_criteria(criteria)
```
Comment (losborne) [inline lib/measure-loader/source_data_criteria_loader.rb]:
```suggestion
    def extract_fields_from_single_code_reference_data_criteria(criteria)
```

File: test/unit/measure-loader/cql_loader_test.rb [hunk 7]
```diff
@@ -119,19 +119,40 @@ def test_definition_with_same_name_as_a_library_definition
...
+  def test_source_data_criteria_creation
+    VCR.use_cassette('measure__source_data_criteria_creation', @vcr_options) do
+      measure_file = File.new File.join(@fixtures_path, 'CMS134v6.zip')
+      value_set_loader = Measures::VSACValueSetLoader.new(options: @vsac_options, ticket_granting_ticket: get_ticket_granting_ticket_using_env_vars)
+      loader = Measures::CqlLoader.new(measure_file, @measure_details, value_set_loader)
+      measures = loader.extract_measures
+      assert_equal 1, measures.length
+      measure = measures[0]
+
+      assert_equal measure.source_data_criteria.map(&:hqmfTitle), ["Encounter, Performed", "Procedure, Performed", "Patient Characteristic Sex", "Medication, Active", "Diagnosis", "Intervention, Order", "Intervention, Performed", "Patient Characteristic Race", "Patient Characteristic Payer", "Patient Characteristic Ethnicity", "Laboratory Test, Performed"]
```
Comment (losborne) [inline test/unit/measure-loader/cql_loader_test.rb]: Fixes for Metrics/LineLength and Style/StringLiterals: 

```suggestion
      assert_equal measure.source_data_criteria.map(&:hqmfTitle), [
          'Encounter, Performed',
          'Procedure, Performed',
          'Patient Characteristic Sex',
          'Medication, Active',
          'Diagnosis',
          'Intervention, Order',
          'Intervention, Performed',
          'Patient Characteristic Race',
          'Patient Characteristic Payer',
          'Patient Characteristic Ethnicity',
          'Laboratory Test, Performed'
          ]
```

File: test/unit/measure-loader/cql_loader_test.rb [hunk 8]
```diff
@@ -119,19 +119,40 @@ def test_definition_with_same_name_as_a_library_definition
...
+      assert_equal measure.source_data_criteria[0].description, "Encounter, Performed: Face-to-FaceInteraction"
```
Comment (losborne) [inline test/unit/measure-loader/cql_loader_test.rb]:
```suggestion
      assert_equal measure.source_data_criteria[0].description, 'Encounter, Performed: Face-to-FaceInteraction'
```

File: test/unit/measure-loader/cql_loader_test.rb [hunk 9]
```diff
@@ -119,19 +119,40 @@ def test_definition_with_same_name_as_a_library_definition
...
+      assert_equal measure.source_data_criteria[0].codeListId, "2.16.840.1.113883.3.464.1003.101.12.1048"
```
Comment (losborne) [inline test/unit/measure-loader/cql_loader_test.rb]:
```suggestion
      assert_equal measure.source_data_criteria[0].codeListId, '2.16.840.1.113883.3.464.1003.101.12.1048'
```

File: test/unit/measure-loader/cql_loader_test.rb [hunk 10]
```diff
@@ -119,19 +119,40 @@ def test_definition_with_same_name_as_a_library_definition
...
+      assert_equal measure.source_data_criteria[0].hqmfOid, "2.16.840.1.113883.10.20.28.4.5"
```
Comment (losborne) [inline test/unit/measure-loader/cql_loader_test.rb]:
```suggestion
      assert_equal measure.source_data_criteria[0].hqmfOid, '2.16.840.1.113883.10.20.28.4.5'
```

File: test/unit/measure-loader/cql_loader_test.rb [hunk 11]
```diff
@@ -119,19 +119,40 @@ def test_definition_with_same_name_as_a_library_definition
...
+
+      # Test direct reference code elements are filled with info from hitting vsac
+      assert_equal measure.source_data_criteria[10].description, "Laboratory Test, Performed: UrineProteinTests"
```
Comment (losborne) [inline test/unit/measure-loader/cql_loader_test.rb]:
```suggestion
      assert_equal measure.source_data_criteria[10].description, 'Laboratory Test, Performed: UrineProteinTests'
```

File: test/unit/measure-loader/cql_loader_test.rb [hunk 12]
```diff
@@ -119,19 +119,40 @@ def test_definition_with_same_name_as_a_library_definition
...
+      assert_equal measure.source_data_criteria[10].codeListId, "2.16.840.1.113883.3.464.1003.109.12.1024"
```
Comment (losborne) [inline test/unit/measure-loader/cql_loader_test.rb]:
```suggestion
      assert_equal measure.source_data_criteria[10].codeListId, '2.16.840.1.113883.3.464.1003.109.12.1024'
```

File: test/unit/measure-loader/cql_loader_test.rb [hunk 13]
```diff
@@ -119,19 +119,40 @@ def test_definition_with_same_name_as_a_library_definition
...
+      assert_equal measure.source_data_criteria[4].description, "Diagnosis: KidneyFailure"
```
Comment (losborne) [inline test/unit/measure-loader/cql_loader_test.rb]:
```suggestion
      assert_equal measure.source_data_criteria[4].description, 'Diagnosis: KidneyFailure'
```

File: test/unit/measure-loader/cql_loader_test.rb [hunk 14]
```diff
@@ -119,19 +119,40 @@ def test_definition_with_same_name_as_a_library_definition
...
+      assert_equal measure.source_data_criteria[4].codeListId, "2.16.840.1.113883.3.464.1003.109.12.1028"
```
Comment (losborne) [inline test/unit/measure-loader/cql_loader_test.rb]:
```suggestion
      assert_equal measure.source_data_criteria[4].codeListId, '2.16.840.1.113883.3.464.1003.109.12.1028'
```

File: test/unit/measure-loader/cql_loader_test.rb [hunk 15]
```diff
@@ -176,16 +197,16 @@ def test_measure_including_draft
...
       value_set_loader = Measures::VSACValueSetLoader.new(options: @vsac_options_w_draft, ticket_granting_ticket: get_ticket_granting_ticket_using_env_vars)
       loader = Measures::CqlLoader.new(measure_file, @measure_details, value_set_loader)
       measures = loader.extract_measures
       measure = measures[0]
 
-      assert_equal "BCSTest", measure.title
-      assert_equal "40280582-57B5-1CC0-0157-B53816CC0046", measure.hqmf_id
-      assert_equal 1, measure.population_sets.size
-      assert_equal 4, measure.population_criteria.keys.count
+      assert_equal "Initiation and Engagement of Alcohol and Other Drug Dependence Treatment", measure.title
```
Comment (losborne) [inline test/unit/measure-loader/cql_loader_test.rb]:
```suggestion
      assert_equal 'Initiation and Engagement of Alcohol and Other Drug Dependence Treatment', measure.title
```

File: test/unit/measure-loader/cql_loader_test.rb [hunk 16]
```diff
@@ -176,16 +197,16 @@ def test_measure_including_draft
...
+      assert_equal "40280382-6258-7581-0162-92A37A9B15DF", measure.hqmf_id
```
Comment (losborne) [inline test/unit/measure-loader/cql_loader_test.rb]:
```suggestion
      assert_equal '40280382-6258-7581-0162-92A37A9B15DF', measure.hqmf_id
```
